compiler/include/baseAST.h

```diff
@@ -392,6 +398,25 @@ static inline const LcnSymbol* toConstLcnSymbol(const BaseAST* a)
   return isLcnSymbol(a) ? (const LcnSymbol*) a : NULL;
 }
 
+CallExpr* getDesignatedCall(ContextCallExpr* a);
+
+static inline CallExpr* toCallExpr(BaseAST* a)
+{
+  if (!a) return NULL;
+  if (a->astTag == E_CallExpr) return (CallExpr*) a;
+  if (a->astTag == E_ContextCallExpr) return getDesignatedCall((ContextCallExpr*)a);
+  return NULL;
+}
+
+static inline const CallExpr* toConstCallExpr(const BaseAST* a)
+{
+  if (!a) return NULL;
+  if (a->astTag == E_CallExpr) return (const CallExpr*) a;
+  if (a->astTag == E_ContextCallExpr) return getDesignatedCall((ContextCallExpr*)a);
+  return NULL;
+}
```

Comment:
Given the name of this function, I'd expect it to return the ConstRef call, not the Ref call.
Can the inconsistency be explained or eliminated?